lib/postgrex/pgpass.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Postgrex.Pgpass do
+  @moduledoc """
+  Parses .pgpass file to obtain database credentials
+  """
+
+  @pgpass_path Application.get_env(:postgrex, :pgpass, Path.join(System.user_home, '.pgpass'))
+  use Bitwise, only_operators: true
+
+  @doc """
+  Obtains the username given a keyword list containing hostname, database & port
+  """
+  def username(opts) do
+    with [_h, _p, _d, username, _pwd] <- pgpass_for(opts[:hostname], opts[:database], opts[:port]) do
+      username
+    else
+      _ -> nil
+    end
+  end
+
+  @doc """
+  Obtains the password given a keyword list containing hostname, database, port and optionally the username.
+  """
+  def password(opts) do
+    with [_h, _p, _d, _u, password] <- pgpass_for(opts[:hostname], opts[:database], opts[:port], opts[:username]) do
+      password
+    else
+      _ -> nil
+    end
+  end
+
+  defp pgpass_for(hostname, database, port, username \\ nil) do
+    with true <- readable_pgpass_file?(),
+         {:ok, contents} <- File.read(@pgpass_path),
+         entries when is_list(entries) <- parse(contents) do
+      entries
+      |> filter_hostname(hostname)
+      |> filter_port(port)
+      |> filter_database(database)
+      |> filter_username(username)
+      |> List.first
+    end
+  end
+
+  defp parse(contents) do
+    contents
+    |> String.split("\n")
+    |> Enum.reject(&(String.match?(&1,~r/^#/)))
+    |> Enum.map(&(String.split(&1, ":")))
+  end
+
+  defp readable_pgpass_file? do
+    with {:ok, stat}     <- File.stat(@pgpass_path),
+    do: (stat.mode &&& 0o0777) == 0o0600 || Mix.env == :test,
```

Comment:
I commented about that below. I can't figure out how to get git(GitHub) to preserve the permissions. Should I just chmod the support file in the unit's setup block?